spec.html

```diff
@@ -15692,21 +15788,77 @@ <h1>Runtime Semantics: BindingInitialization</h1>
+        </emu-grammar>
+        <emu-alg>
+          1. Let _excludedNames_ be the result of performing PropertyBindingInitialization for |BindingPropertyList| using _value_ and _environment_ as the argument.
+          1. ReturnIfAbrupt(_excludedNames_).
+          1. Return NormalCompletion(~empty~).
+        </emu-alg>
+
+        <emu-grammar>ObjectBindingPattern : `{` BindingRestProperty `}`</emu-grammar>
+        <emu-alg>
+          1. Let _excludedNames_ be a new empty List.
+          1. Return the result of performing RestBindingInitialization of _BindingRestProperty_ with _value_, _environment_ and _excludedNames_ as the arguments.
+        </emu-alg>
+
+        <emu-grammar>ObjectBindingPattern : `{` BindingPropertyList `,` BindingRestProperty `}`</emu-grammar>
+        <emu-alg>
+          1. Let _excludedNames_ be the result of performing PropertyBindingInitialization of |BindingPropertyList| using _value_ and _environment_ as arguments.
+          1. ReturnIfAbrupt(_excludedNames_).
+          1. Return the result of performing RestBindingInitialization of _BindingRestProperty_ with _value_, _environment_ and _excludedNames_ as the arguments.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="Rest-RuntimeSemantics-PropertyBindingInitialization">
+        <h1>Runtime Semantics: PropertyBindingInitialization</h1>
+        <p>With parameters _value_ and _environment_.</p>
+
+        <emu-note>These collect a list of all bound property names rather than just empty completion.</emu-note>
+
         <emu-grammar>BindingPropertyList : BindingPropertyList `,` BindingProperty</emu-grammar>
         <emu-alg>
-          1. Perform ? BindingInitialization for |BindingPropertyList| using _value_ and _environment_ as arguments.
-          1. Return the result of performing BindingInitialization for |BindingProperty| using _value_ and _environment_ as arguments.
+          1. Let _boundNames_ be the result of performing PropertyBindingInitialization for |BindingPropertyList| using _value_ and _environment_ as arguments.
+          1. ReturnIfAbrupt(_boundNames_).
+          1. Let _nextNames_ be the result of performing PropertyBindingInitialization for |BindingProperty| using _value_ and _environment_ as arguments.
+          1. ReturnIfAbrupt(_nextNames_).
+          1. Append each item in _nextNames_ to the end of _boundNames_.
+          1. Return _boundNames_.
         </emu-alg>
+
         <emu-grammar>BindingProperty : SingleNameBinding</emu-grammar>
         <emu-alg>
           1. Let _name_ be the string that is the only element of BoundNames of |SingleNameBinding|.
-          1. Return the result of performing KeyedBindingInitialization for |SingleNameBinding| using _value_, _environment_, and _name_ as the arguments.
+          1. Let _status_ be the result of performing KeyedBindingInitialization for |SingleNameBinding| using _value_, _environment_, and _name_ as the arguments.
+          1. ReturnIfAbrupt(_status_).
```

Comment:
If you're doing these conversions to `?` syntax, you might be able to apply it here too, and use `Perform ? KeyBindingInitialization for |SingleNameBinding| using _value_, _environment_, and _name_ as the arguments.` There are some other uses of ReturnIfAbrupt in this PR that you could remove similarly. However, the current spec seems to have some uses of `?` and some ReturnIfAbrupt in a way that I don't really understand; maybe these uses are not allowed for some reason.

Reply:
I've changed any `ReturnIfAbrupt` where the `Let` value isn't used. In other words, now the only usage of `ReturnIfAbrupt` is where the `Let` is used later in the steps.